src/main/java/org/testng/xml/XmlTest.java

```diff
@@ -582,7 +582,7 @@ public String toXml(String indent) {
   @Override
   public String toString() {
 //    return toXml("");
-    StringBuffer result = new StringBuffer("[Test: \"" + m_name + "\"")
+    StringBuilder result = new StringBuilder("[Test: \"" + m_name + "\"")
```

Comment:
Yes, it makes sense and I have no opinion about `String#format` vs `StringBuilder`.
I think it is just sad to have a pull-request which modifies lines and keeps "bad" code ;)